src/ucp/rndv/rndv_put.c

```diff
@@ -637,6 +638,8 @@ ucp_proto_rndv_put_mtype_probe(const ucp_proto_init_params_t *init_params)
         comp_cb = ucp_proto_rndv_put_mtype_completion;
     }
 
+    ucp_proto_rndv_mtype_update_sys_dev(init_params, &frag_mem_info);
```

Comment:
maybe need to update/modify ucp_mm_get_alloc_md_index function? it seems disconnected to do it here